sstables/storage.cc

```diff
@@ -452,6 +455,16 @@ class s3_storage : public sstables::storage {
     }
 
     virtual sstring prefix() const override { return _location; }
+    virtual sstring location(const sstable& sst, std::optional<component_type> f = std::nullopt) const override {
```

Comment:
That's good question. The call should return some string that a human can read and understand what sstable the log string is talking about. In case of a local file it's a path (and some tests do use it as a _path_, but it's OK for a test). For S3 it's rather an s3 object path which's not the same as legacy file path even relative to the bucket (and sometimes it can be missing because uuid is not allocated, but you're fixing it already). For double-backed storage it can be anything suitable.

On the other hand, there's #12707 that tries to keep "location" on a table and make it abstract enough not to make sstables layer care about it at all